docs/manual/en/manual.txt

```diff
@@ -0,0 +1,258 @@
+
+.Emulators
+image::../resources/emulators.png[200,200]
+
+- DOSBox OPL: This emulator is a balanced choice, producing good fidelity sound while using a very reasonable quantity of CPU resources.
+- Nuked OPL: This emulator offers excellent fidelity, aiming to provide nearly cycle-exact emulation. However, CPU usage is significant, so it is only recommended with a small number of chips, or for non-realtime rendering.
+
+=== Operators
+
+The four operators, whether functioning as modulator or carrier, provide the same parameters.
+
+.An operator
+image::../resources/operator.png[270,270]
+
+==== Waveforms
+
+The first parameter is the oscillator waveform, located in the lower left corner. One of 8 waveforms may be sellected.
+
+
+. sine
+. half-sine
+. absolute sine
+. quarter-sine, also known as pulse sine
+. alternating sine
+. camel sine, also known as absolute alternating sine
+. square
+. logarithmic sawtooth, also known as derived square
+
+==== Envelopes
+
+The envelope generator is controlled by four dials labelled "ADSR".
+The envelope controls the amplitude (volume) of an oscillator with respect to time.
+
+.ADSR Envelope
+image::../resources/adsr.svg[250,250]
+
+The attack (A) and decay (D) parameters control the durations of those envelope stages.
+
+The sustain (S) parameter controls the volume level held after A + D has elapsed, and as long as the key remains pressed.
+However, if the "Sustain" flag is disabled, there is no sustain stage, and the envelope effectively becomes an "ADR".
+
+The release (R) parameter comes after sustain, and similarly to "A" and "D", specifies the duration of the release stage.
+
+==== Flags
+
+The "Tremolo" and "Vibrato" flags apply small modulations to the operator, affecting volume and frequency respectively.
+
+"Sustain" enables or disables the "S" stage of the envelope. Refer to the explanation on envelopes.
+
+"Key scaling" automatically shortens the duration of the envelope as the pitch of the note increases.
+
+==== Levels
+
+The output level "Lv" is the global volume of the operator.
+Note that there is no "0" setting; even at the lowest level, the operator can be heard faintly.
+
+The frequency multiplier "F*" controls the base frequency of the operator, in octaves.
+
+The key scale level "Ksl" determines the extent to which amplitude is attenuated as pitch increases.
+
+=== Other parameters
+
+The remaining parameters fall into two groups: Miscellaneous and Global.
+
+A miscellaneous parameter applies to the currently selected program in the editor. On the other hand, a global parameter is a property of the bank, and applies to all programs in that bank.
+
+.Miscellaneous parameters
+image::../resources/misc.png[200,200]
+
+"Velocity Offset" is a fixed modifier applied to the velocity of MIDI notes, and can be set in the range [0, 127]. A positive offset increases velocity, and a negative value decreases it.
```

Comment:
Should the range be [-127, 127]?

Reply:
Yes this is a mistake, it's the signed 8 bit integer range,